components/button/button-add.js

```diff
@@ -59,30 +79,131 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
 			button:${unsafeCSS(getFocusPseudoClass())} .line {
 				border-top-color: var(--d2l-color-celestine);
 			}
+		`;
+	}
 
-			.content {
+	constructor() {
+		super();
+
+		this.hintNearbyDistance = DEFAULT_HINT_DISTANCE;
+		this.hintTiming = HINT_TIMING.ALWAYS;
+		this.textVisible = false;
+
+		this._buttonId = getUniqueId();
+		this._hasFocusOrHover = false;
+		this._visibleDistance = 0;
+
+		this.addEventListener('focus', this._onFocusHover);
+		this.addEventListener('mouseover', this._onFocusHover);
+		this.addEventListener('blur', this._onBlur);
```

Comment:
Nit: if we want to share handlers for focus and mouse events, maybe we should name them... `_onFocusMouseOver` and `_onBlurMouseOut`?